component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteStore.java

```diff
@@ -0,0 +1,1183 @@
+                condition = this.replaceConditionWithParameter(condition, findConditionParameterMap, constantMap);
+                readQuery.append(condition);
+            }
+            for (Attribute attribute : this.attributes) {
+                attributeList.add(attribute.getName());
+            }
+            statement = connection.prepareStatement(readQuery.toString());
+            resultSet = statement.executeQuery();
+            return new ApacheIgniteIterator(statement, resultSet, this.tableName,
+                    attributeList, this.attributes);
+        } catch (SQLException e) {
+            ApacheIgniteTableUtils.cleanup(resultSet, statement);
+            if (e.getSQLState().equals("08006")) {
+                throw new ConnectionUnavailableException("Error retrieving records from the table  " + this.tableName +
+                        " in ignite store : " + e.getMessage(), e);
+            }
+            throw new ApacheIgniteTableException("Error retrieving records from the table  " + this.tableName +
+                    " in ignite store : " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Check if matching record exist or not
+     *
+     * @param containsConditionParameterMap map of matching StreamVariable Ids and their values corresponding to the
+     *                                      compiled condition
+     * @param compiledCondition             the compiledCondition against which records should be matched
+     * @return if matching record found or not
+     */
+    @Override
+    protected boolean contains(Map<String, Object> containsConditionParameterMap,
+                               CompiledCondition compiledCondition) throws ConnectionUnavailableException {
+
+        ApacheIgniteCompiledCondition igniteCompiledCondition = (ApacheIgniteCompiledCondition) compiledCondition;
+        String condition = igniteCompiledCondition.getCompiledQuery();
+        Map<Integer, Object> constantMap = igniteCompiledCondition.getParameterConstants();
+        ResultSet rs = null;
+        PreparedStatement statement = null;
+        StringBuilder readQuery = new StringBuilder();
+        readQuery.append(SELECT).append(WHITESPACE).append(ASTERISK).append(WHITESPACE).append(FROM).append(WHITESPACE)
+                .append(this.tableName);
+        try {
+            if (!condition.equals(HASH)) {
+                readQuery.append(WHITESPACE).append(WHERE).append(WHITESPACE);
+                condition = this.replaceConditionWithParameter(condition, containsConditionParameterMap, constantMap);
+                readQuery.append(condition);
+            }
+            statement = connection.prepareStatement(readQuery.toString());
+            rs = statement.executeQuery();
+            return rs.next();
+        } catch (SQLException e) {
+            if (e.getSQLState().equals("08006")) {
+                throw new ConnectionUnavailableException("Error performing contains check for ignite store '"
```

Comment:
add siddhi app name in the error msg if possible. give as much context as possible with each error message